### ISSUE TITLE
Pre-process move-original fails with "Conversion between raster formats not supported"

### ISSUE THREAD

**Issue by NoelAbrahams** [blitline-dev/simple_blitline_node#15]
Hi,

We have been using the pre-process move-original feature in order to store the original image.

The source images are of various mime types, but we are attempting to save them all as `.jpg`.

```json
{
  "application_id": "YOUR_APP_ID",
  "src": "http://example.com/image.png",
  "pre_process": {
    "move_original": {
      "azure_destination": {
        "account_name": "account",
        "shared_access_signature": "https://foo.blob.core.windows.net/sample/image.jpg"
      }
    }
  }
}
```

This works fine when the source image is `.jpg`, but fails for other types such as `.png` or `.gif`.

The error reported is

> Image processing failed. Conversion between raster formats not supported. Please just output via job functions and save

or

> Image processing failed. .gif -> .jpg conversion unsupported here

Incidentally, using a the `no_op` function to store the original ***does*** work.

I would expect the `move-original` feature to work the same.

Thanks.

**Comment by blitline-dev** [comment]
Yes, unfortunately this is a limitation of the Blitline process. Move original doesn’t convert between formats, it literally just moves a file. The file is never opened, and there is no raster image processing that happens through that code path.

No-op DOES go though image processing, which is why the conversion between formats is possible.

-- 
Blitline Support
Sent with Airmail

On November 7, 2016 at 12:43:49 AM, nabog (notifications@github.com) wrote:

Hi,

We have been using the pre-process move-original feature in order to store the original image.

The source images are of various mime types, but we are attempting to save them all as .jpg.

{
  "application_id": "YOUR_APP_ID",
  "src": "http://example.com/image.png",
  "pre_process": {
    "move_original": {
      "azure_destination": {
        "account_name": "account",
        "shared_access_signature": "https://foo.blob.core.windows.net/sample/image.jpg"
      }
    }
  }
}
This works fine when the source image is .jpg, but fails for other types such as .png or .gif.

The error reported is

Image processing failed. Conversion between raster formats not supported. Please just output via job functions and save

or

Image processing failed. .gif -> .jpg conversion unsupported here

Incidentally, using a the no_op function to store the original does work.

I would expect the move-original feature to work the same.

Thanks.
